### ISSUE TITLE
0AD 0.23 fails to build 

### ISSUE THREAD

**Issue by strycore** [lutris/buildbot#47]
Tested on the 16.04 buildbot container with current build script

```
ModIo.cpp
../../../source/ps/ModIo.cpp: In constructor ‘ModIo::ModIo()’:
../../../source/ps/ModIo.cpp:120:113: error: ‘sodium_base64_VARIANT_ORIGINAL’ was not declared in this scope
  if (sodium_base642bin((unsigned char*)&m_pk, sizeof m_pk, pk_str.c_str(), pk_str.size(), NULL, &bin_len, NULL, sodium_base64_VARIANT_ORIGINAL) != 0 || bin_len != sizeof m_pk)
                                                                                                                 ^
../../../source/ps/ModIo.cpp:120:143: error: ‘sodium_base642bin’ was not declared in this scope
  if (sodium_base642bin((unsigned char*)&m_pk, sizeof m_pk, pk_str.c_str(), pk_str.size(), NULL, &bin_len, NULL, sodium_base64_VARIANT_ORIGINAL) != 0 || bin_len != sizeof m_pk)
                                                                                                                                               ^
../../../source/ps/ModIo.cpp: In static member function ‘static bool ModIo::ParseSignature(const std::vector<std::__cxx11::basic_string<char> >&, SigStruct&, const PKStruct&, std::__cxx11::string&)’:
../../../source/ps/ModIo.cpp:777:114: error: ‘sodium_base64_VARIANT_ORIGINAL’ was not declared in this scope
   if (sodium_base642bin((unsigned char*)&sig, sizeof sig, msg_sig.c_str(), msg_sig.size(), NULL, &bin_len, NULL, sodium_base64_VARIANT_ORIGINAL) != 0 || bin_len != sizeof sig)
                                                                                                                  ^
../../../source/ps/ModIo.cpp:777:144: error: ‘sodium_base642bin’ was not declared in this scope
   if (sodium_base642bin((unsigned char*)&sig, sizeof sig, msg_sig.c_str(), msg_sig.size(), NULL, &bin_len, NULL, sodium_base64_VARIANT_ORIGINAL) != 0 || bin_len != sizeof sig)
                                                                                                                                                ^
../../../source/ps/ModIo.cpp:793:121: error: ‘sodium_base64_VARIANT_ORIGINAL’ was not declared in this scope
   if (sodium_base642bin(global_sig, sizeof global_sig, sig_lines[3].c_str(), sig_lines[3].size(), NULL, &bin_len, NULL, sodium_base64_VARIANT_ORIGINAL) != 0 || bin_len != sizeof global_sig)
                                                                                                                         ^
../../../source/ps/ModIo.cpp:793:151: error: ‘sodium_base642bin’ was not declared in this scope
   if (sodium_base642bin(global_sig, sizeof global_sig, sig_lines[3].c_str(), sig_lines[3].size(), NULL, &bin_len, NULL, sodium_base64_VARIANT_ORIGINAL) != 0 || bin_len != sizeof global_sig)
                                                                                                                                                       ^
engine.make:555: recipe for target 'obj/engine_Release/ModIo.o' failed
make[1]: *** [obj/engine_Release/ModIo.o] Error 1
make[1]: *** Waiting for unfinished jobs....
Makefile:110: recipe for target 'engine' failed
```

**Comment by r2d02** [comment]
Same issue here

**Comment by DevPGSV** [comment]
I was able to solve this.
I am running debian stretch.

The step I did to be able to solve this problem:


1. Add stretch-backports ([how to](https://backports.debian.org/Instructions/))
2. `apt-get -t stretch-backports install libsodium-dev`
3. Retry compilation

The problem is:
In debian stretch the available libsodium-dev package is on version 1.0.11: [packgae](https://packages.debian.org/stretch/libsodium-dev)

However the 0AD code requires a newer version of libsodium. I don't know exactly what version is required, what I know is that the [line 64 of sodium/utils.h](https://github.com/jedisct1/libsodium/blob/master/src/libsodium/include/sodium/utils.h#L67) has to define sodium_base64_VARIANT_ORIGINAL, and in version 1.0.11 that is not there.
